scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,680 @@
+    # They should be listed in the `unix:examples.file` configuration line
+    if [[ $(grep unix:examples.files "$md_build/skyscraper.pro" 2>/dev/null | cut -d= -f2-) ]]; then
+
+        local files=$(grep unix:examples.files "$md_build/skyscraper.pro" | cut -d= -f2-)
+
+        for file in $files; do
+            # Copy the files to the configuration folder. Skip config.ini and artwork.xml, they're already handled
+            if [[ $file != "artwork.xml"  && $file != "config.ini" ]]; then
+                cp -f "$md_build/$file" "$md_conf_dir"
+            fi
+        done
+
+    else
+
+        # Fallback to the known resource files list
+        cp -f "$md_build/artwork.xml.example"* "$md_conf_dir"
+
+        # Copy resources
+        for resource_file in ARTWORK.md README.md mameMap.csv tgdb_developers.json tgdb_publishers.json hints.txt; do
+            cp -f "$md_build/$resource_file" "$md_conf_dir"
+        done
+
+    fi
+
+    # Copy the rest of the folders
+    cp -rf "$md_build/resources" "$md_conf_dir"
+
+    # Create the import folders
+    for folder in boxart snaps covers marquees screenshots textual videos wheels; do
+        mkUserDir "$md_conf_dir/import/$folder"
+    done
+
+    # Create the LocalDB cache folder
+    mkUserDir "$md_conf_dir/dbs"
+
+}
+
+# Scrape one system, passed as parameter
+function scrape_skyscraper() {
+    local system="$1"
+    [[ -z "$system" ]] && return
+
+    iniConfig " = " '"' "$configdir/all/skyscraper.cfg"
+    eval $(_load_config_skyscraper)
+
+    local gamelist
+    local media_dir
+
+    if [[ "$use_rom_folder" -eq 1 ]]; then
+        gamelist="$romdir/$system"
+        media_dir="$romdir/$system/media"
+        # If we're saving to the ROM folder, then use relative paths in the gamelist
+        relative="--relative"
+    else
+        gamelist="$home/.emulationstation/gamelists/$system"
+        media_dir="$home/.emulationstation/downloaded_media/$system"
+    fi
+
+    local -a params=("--unattend" "--skipped")
+
+    params+=(-o "$media_dir")
+    params+=(-g "$gamelist" "$relative")
+    params+=(-p "$system")
+    params+=(-s "$scrape_source")
+
+    if [[ "$download_videos" -eq 1 ]]; then
+        params+=(--videos)
+    fi
+
+    if [[ "$cache_marquees" -eq 0 ]]; then
+        params+=(--nomarquees)
+    fi
+
+    if [[ "$cache_covers" -eq 0 ]]; then
+        params+=(--nocovers)
+    fi
+
+    if [[ "$cache_screenshots" -eq 0 ]]; then
+        params+=(--noscreenshots)
+    fi
+
+    if [[ "$cache_wheels" -eq 0 ]]; then
+        params+=(--nowheels)
+    fi
+
+    if [[ "$rom_name" -eq 1 ]]; then
+        params+=(--forcefilename)
+    fi
+
+    if [[ "$remove_brackets" -eq 1 ]]; then
+        params+=(--nobrackets)
+    fi
+
+    if [[ "$force_refresh" -eq 1 ]]; then
+        params+=(--refresh)
+    fi
+
+    # trap ctrl+c and return if pressed (rather than exiting retropie-setup etc)
+    trap 'trap 2; return 1' INT
+        sudo -u $user "$md_inst/Skyscraper" "${params[@]}"
```

Comment:
why sudo ?

Reply:
That was the original code in the `scraper` module (https://github.com/RetroPie/RetroPie-Setup/blob/13b99f039c2dafa801593c5afd712ed6a32600a1/scriptmodules/supplementary/scraper.sh#L136). I assume because the RetroPie-Setup runs as `root`, but the scraping is done by the user.

Reply:
Ah yes. forgot about that.